lib/vrl/compiler/src/expression.rs

```diff
@@ -134,25 +134,49 @@ impl Expr {
         }
     }
 
+    pub fn as_literal(&self, keyword: &'static str) -> Result<Value, super::function::Error> {
+        let literal = match self {
+            Expr::Literal(literal) => Ok(literal.clone()),
+            Expr::Variable(var) if var.value().is_some() => {
+                match var.value().unwrap().clone().into() {
```

Comment:
Suggestion to circumvent the `unwrap`:

```suggestion
            Expr::Variable(Variable {
                value: Some(value), ..
            }) => match value.clone().into() {
```

Only works if that field is `pub(crate)` though.

Reply:
This code is largely copied from the AST function args, which does a similar thing. I think there was a reason why we were keen to avoid exposing too much of the internals - hence the adopted solution here..